src/BenchmarkDotNet/Validators/BenchmarkClassValidator.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using BenchmarkDotNet.Extensions;
+using BenchmarkDotNet.Running;
+
+namespace BenchmarkDotNet.Validators
+{
+    public class BenchmarkClassValidator: IValidator
+    {
+        public static readonly IValidator Default = new BenchmarkClassValidator();
+        
+        public bool TreatsWarningsAsErrors => false;
+
+        private BenchmarkClassValidator() { }
+        
+        public IEnumerable<ValidationError> Validate(ValidationParameters validationParameters)
+        {
+            var notValid = validationParameters.Benchmarks.Where(x => x.Target.Type.IsGenericType)
+                                                          .Where(HasPrivateGenericArguments).ToList();
```

Comment:
nit: `.ToList()` is not mandatory here, you can just yield directly from the `foreach` loop or return and `IEnumberable`

```cs
=> validationParameters.Benchmarks
       .Where(x => x.Target.Type.IsGenericType && HasPrivateGenericArguments(x.Target.Type))
       .Select(benchmark => new ValidationError(true, $"Generic class {benchmark.Target.Type.GetDisplayName()} has non public generic argument(s)"));
```